### PR TITLE
Fix: sdk7 text shape alignment

### LINKED ISSUE #879: TextShapes broken format
### **Issue Description:**

It seems all entities with a TextShape have a broken format, they end up with 1 character per line, and some strings are completely out of place.

### **Steps to reproduce:**

  1. Go to dclexperience.dcl.eth
  2. Look at the screens up in the clouds, all of the text in those 3D UIs is either missing or broken

You can also observe the same problem in several scenes in Goerli Plaza
for example
https://github.com/decentraland/sdk7-goerli-plaza/tree/main/BasicInteractions
https://github.com/decentraland/sdk7-goerli-plaza/tree/main/Jukebox
https://github.com/decentraland/sdk7-goerli-plaza/tree/main/advanced-avatar-swap

### **Expected behaviour:**

The text should be readable, and displayed horizontally

### **Current behaviour:**

Text appears as 1 character per line

### **Reproduction rate:**
- [ ] < 10%
- [ ]  < = 25%
- [ ]  50%
- [ ]  < = 75%
- [ x ]  100%

### **Severity:**

- [ ] Cosmetic
- [ ] Minor
- [ ] Medium
- [ x ] High
- [ ] Critical
- [ ] Blocker

### **Priority:**
- [ ] High
- [ ] Medium
- [ ] Low

### **Platforms / OS:**
- [ x ] Windows 11
- [ ] Windows 10
- [ ] Mac Sonoma
- [ ] Mac (specify the version on Additional Notes)

### LINKED ISSUE #881: TextShapes where they shouldn't be
### **Issue Description:**

Suddenly I see text shapes that should be up in the clouds in the 3D UI appearing in places where they shouldn't be. Even attached to my avatar.

In this video you can see they appeared where we should have NFTs, and then on my avatar.

For some reason, I couldn't go back to 3rd person view at that time (another bug)

It's unclear how this happened, perhaps related to https://github.com/decentraland/unity-explorer/issues/879
I jumped between worlds a bit before coming back to the plaza, maybe some entitiy ID problem.

### **Steps to reproduce:**

  1. Go to dclexperience.dcl.eth
  2. Go to Maximocossetti.dcl.eth
  3. Grab a beer mug in that scene
  4. Jump back to dclexperience.dcl.eth
  (at this point the beermug in your hand is replaced by text, noticeable only when you walk)
  5. Visit the NFT art gallery just outside the bar

### **Expected behaviour:**

There should be no text in the NFT art gallery outside the bar
There should be no random texts attached to my body
I should be able to leave the 1st person view

### **Current behaviour:**

I'm stuck in 1st person view
I have text in my hand, that I can see swinging when I walk, probably replacing the beer mug I just grabbed
If I look down and jump, I see plenty more text attached to my body
If I walk out to the NFT art gallery, I see some more text on the internal walls, where I should see entities with NFTShape

All of this text seems to come from the events board that should be up in the clouds

### **Reproduction rate:**
- [ ] < 10%
- [ ]  < = 25%
- [ ]  50%
- [ ]  < = 75%
- [ x ]  100%

### **Severity:**

- [ ] Cosmetic
- [ ] Minor
- [ ] Medium
- [ x ] High
- [ ] Critical
- [ ] Blocker

### **Priority:**
- [ ] High
- [ ] Medium
- [ ] Low

### **Platforms / OS:**
- [ x ] Windows 11
- [ ] Windows 10
- [ ] Mac Sonoma
- [ ] Mac (specify the version on Additional Notes)

#### **Evidence:**

https://github.com/decentraland/unity-explorer/assets/3507907/24cb2af1-f194-43cd-ab8d-e7fcae0d0531

#### **Additional Notes:**

I see this in all kinds of places after exploring for a while
my theory is that the TextShape component is not being removed from the entity pool when unloading a scene

### PR DESCRIPTION
**Pull request by popuz**
## What does this PR change?
fix #879, 
fix #881

Main problem was due to the re-used values from the pooled TMPro object transforms, that were never reset.
- replicate most of the unity-renderer logic related to the protobuf values
- replaced logic by the usage of generic pool
- moved some logic into Systems and Extensions
- reused MaterialPropertyBlock 

## How to test the changes?

1. Launch the explorer
2. Goto dclexperience
3. Verify that text is on proper places. For example in Carousel - JUMP-IN, dates, titles
NOTE:  while most of texts will be on the proper place, you will also see text that should be hidden - this will be fixed in another PR
4. Verify you don't see a lot of black/white textures at the bar at 0,0 (same position as in #708)

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by anicalbano — COMMENTED** [review]
❔ The text look ok, taking in consideration what we see that should be hidden. The video shows that.
But regarding the glitches, I can still see them:
image


https://github.com/decentraland/unity-explorer/assets/139157886/7b14d051-d2ee-4f8a-8203-3ab05bff59fa

**Comment by popuz** [comment]
> ❔ The text look ok, taking in consideration what we see that should be hidden. The video shows that. But regarding the glitches, I can still see them: image
> 
>  06.05.2024_12.58.11_REC.mp4

Thanks, Ani. It is enough that there is no text at the glitched object, this object will be removed by content team soon

**Review by anicalbano — APPROVED** [review]

File: Explorer/Assets/DCL/PluginSystem/World/TextShapePlugin.cs [hunk 1]
```diff
@@ -5,55 +5,49 @@
...
     {
-        private readonly ITextShapeRendererFactory textShapeRendererFactory;
         private readonly IPerformanceBudget instantiationFrameTimeBudgetProvider;
         private readonly IComponentPoolsRegistry componentPoolsRegistry;
+        private readonly IFontsStorage fontsStorage;
 
-        public TextShapePlugin(IPerformanceBudget instantiationFrameTimeBudgetProvider, IComponentPoolsRegistry componentPoolsRegistry, IPluginSettingsContainer settingsContainer) : this(
-            instantiationFrameTimeBudgetProvider,
-            componentPoolsRegistry,
-            settingsContainer.GetSettings<FontsSettings>().AsCached()
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/PluginSystem/World/TextShapePlugin.cs:27]: Can you please move `FontsSettings` to the settings of **this** plugin just like it's done for everything else?

File: Explorer/Assets/DCL/PluginSystem/World/TextShapePlugin.cs [hunk 2]
```diff
@@ -5,55 +5,49 @@
...
         }
 
-        public UniTask Initialize(IPluginSettingsContainer container, CancellationToken ct) =>
-            UniTask.CompletedTask;
-
         public void InjectToWorld(ref ArchSystemsWorldBuilder<Arch.Core.World> builder, in ECSWorldInstanceSharedDependencies sharedDependencies, in PersistentEntities persistentEntities, List<IFinalizeWorldSystem> finalizeWorldSystems, List<ISceneIsCurrentListener> sceneIsCurrentListeners)
         {
-            InstantiateTextShapeSystem.InjectToWorld(ref builder, textShapeRendererFactory, instantiationFrameTimeBudgetProvider);
-            UpdateTextShapeSystem.InjectToWorld(ref builder);
+            IComponentPool<TextMeshPro> textMeshProPool = componentPoolsRegistry.GetReferenceTypePool<TextMeshPro>();
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/PluginSystem/World/TextShapePlugin.cs:44]: Don't request the pool second time: `componentPoolsRegistry.AddGameObjectPool<TextMeshPro>();` already returns a reference to the pool: just introduce a `private readonly` field

File: Explorer/Assets/Scripts/ECS/Unity/Visibility/Systems/PrimitivesVisibilitySystem.cs [hunk 3]
```diff
@@ -1,21 +1,17 @@
...
 using ECS.LifeCycle.Components;
-using ECS.Unity.Groups;
 using ECS.Unity.PrimitiveRenderer.Components;
-using ECS.Unity.PrimitiveRenderer.Systems;
 
 namespace ECS.Unity.Visibility.Systems
 {
-    [UpdateInGroup(typeof(ComponentInstantiationGroup))]
-    [UpdateAfter(typeof(InstantiatePrimitiveRenderingSystem))]
+    [UpdateInGroup(typeof(SyncedPresentationSystemGroup))]
```
Comment (mikhail-dcl) [inline Explorer/Assets/Scripts/ECS/Unity/Visibility/Systems/PrimitivesVisibilitySystem.cs:13]:
```suggestion
    [UpdateInGroup(typeof(SyncedSimulationSystemGroup))]
    [UpdateAfter(typeof(ComponentInstantiationGroup))]
```

WDYT?

**Review by mikhail-dcl — CHANGES_REQUESTED** [review]

**Review by Kinerius — APPROVED** [review]

**Review by mikhail-dcl — APPROVED** [review]

**Pull request merged**